node_package/src/serverRenderReactComponent.ts

```diff
@@ -112,8 +111,21 @@ as a renderFunction and not a simple React Function Component.`);
       let promiseResult;
 
       try {
+        const awaitedRenderResult = await renderResult;
+
+        // If replayServerAsyncOperationLogs node renderer config is enabled, the console.history will contain all logs happened during sync and async operations.
+        // If the config is disabled, the console.history will be empty, because it will clear the history after the sync execution.
+        // In case of disabled config, we will use the console.history after sync execution, which contains all logs happened during sync execution.
+        const consoleHistoryAfterAsyncExecution = console.history;
+        let consoleReplayScript = '';
+        if ((consoleHistoryAfterAsyncExecution?.length ?? 0) > (consoleHistoryAfterSyncExecution?.length ?? 0)) {
+          consoleReplayScript = buildConsoleReplay(consoleHistoryAfterAsyncExecution);
+        } else {
+          consoleReplayScript = buildConsoleReplay(consoleHistoryAfterSyncExecution);
+        }
+
         promiseResult = {
-          html: await renderResult,
+          html: awaitedRenderResult,
```

Comment:
Why are we basically just moving the `await` to line 114?

Reply:
The console replay script must be built after awaiting the rendering promise, so it includes all console logs happened during sync and async operations on the server.